### ISSUE TITLE
[SC 2.4.1] Bypass Blocks: navigation mechanisms to skip links (frontend-template)

### ISSUE THREAD

**Issue by armenos** [joomla-projects/accessibility#9]
**Description of the issue** (WCAG 2.0 Level A Issue)
Missing skip-link navigation. 
There is no option available to bypass blocks of content on a Web page. 
You do this with your keyboard [Tab] key. This is a tool/(menu) for people who can not use a mouse. With skip link you can go directly to the content. 
A wide explanation of this issue can be found on the following page. Here you can read how you could solve the problem.
Explanation link: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html 

**What happens?**
If you want to navigate with keyboard quickly to the main content that is not possible now. The website visitor must have an opportunity to skip repetitive information blocks. Which are present on the webpage by using the [Tab] key. You can think of: menus, search box, logo, etc.

**What do you expect**
The best way to skip repetitive content is by providing a skip link on the webpage that leads directly to the main content.
If we want to solve this issue, we have to prevent to create any new accessibility issues. Consider use of color when it receives keyboard focus, use of languages in multi language website
http://webaim.org/techniques/skipnav/ 

**Steps to reproduce**
Go to the website and try to navigate with the [Tab] key. 
**How to fix**
A simple way: The best way to skip repetitive content is by providing a skip link on the page that leads directly to the main content. Make sure the link is visible, when it receives the keyboard focus.

The best method to implement this in joomla (in my opinion): make sure that there is some future which makes possible to activate skipnavigation future and corresponds with active modules when it is needed. For example: skip to mainmenu, skip to content, skip to sidebar. You get the pooint. 
_Exempale websites:_
https://werken.belastingdienst.nl 
https://www.werkenvoornederland.nl
http://www.flevoland.nl 
Nice done: http://www.overijssel.nl/ 
Some information websites: 
http://nemesisdesign.net/blog/accessibility/nice-css-skip-links-appearing-focus/
Information on success criteria:
https://www.w3.org/TR/WCAG20-TECHS/G1.html 
https://www.w3.org/WAI/WCAG20/quickref/#navigation-mechanisms-skip 
The links below should help the people involved to test and add the necessary code.
https://www.w3.org/WAI/intro/wcag20 / (Dutch translation: https://www.w3.org/Translations/WCAG20-nl)
https://www.w3.org/WAI/WCAG20/quickref 
https://www.w3.org/WAI/WCAG20/wcag2faq.html 

This is my first post, I did my best to be as clear as possible.
English is not my native language so I hope this is okay translated. I have not added any images and code samples. I like to hear improvements for this post, let me know what information you extra need or miss.

**Comment by brianteeman** [comment]
I assume you are talking about the front end templates and not the admin ?

On 30 May 2016 at 20:22, Armen Mnatsian <notifications@github.com> wrote:

> *Description of the issue* (WCAG 2.0 Level A Issue)
> Missing skip-link navigation.
> There is no option available to bypass blocks of content on a Web page.
> You do this with your keyboard [Tab] key. This is a tool/(menu) for people
> who can not use a mouse. With skip link you can go directly to the content.
> A wide explanation of this issue can be found on the following page. Here
> you can read how you could solve the problem.
> Explanation link:
> https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html
>
> *What happens?*
> If you want to navigate with keyboard quickly to the main content that is
> not possible now. The website visitor must have an opportunity to skip
> repetitive information blocks. Which are present on the webpage by using
> the [Tab] key. You can think of: menus, search box, logo, etc.
>
> *What do you expect*
> The best way to skip repetitive content is by providing a skip link on the
> webpage that leads directly to the main content.
> If we want to solve this issue, we have to prevent to create any new
> accessibility issues. Consider use of color when it receives keyboard
> focus, use of languages in multi language website
> http://webaim.org/techniques/skipnav/
>
> *Steps to reproduce*
> Go to the website and try to navigate with the [Tab] key.
> *How to fix*
> A simple way: The best way to skip repetitive content is by providing a
> skip link on the page that leads directly to the main content. Make sure
> the link is visible, when it receives the keyboard focus.
>
> The best method to implement this in joomla (in my opinion): make sure
> that there is some future which makes possible to activate skipnavigation
> future and corresponds with active modules when it is needed. For example:
> skip to mainmenu, skip to content, skip to sidebar. You get the pooint.
> *Exempale websites:*
> https://werken.belastingdienst.nl
> https://www.werkenvoornederland.nl
> http://www.flevoland.nl
> Nice done: http://www.overijssel.nl/
> Some information websites:
>
> http://nemesisdesign.net/blog/accessibility/nice-css-skip-links-appearing-focus/
> Information on success criteria:
> https://www.w3.org/TR/WCAG20-TECHS/G1.html
> https://www.w3.org/WAI/WCAG20/quickref/#navigation-mechanisms-skip
> The links below should help the people involved to test and add the
> necessary code.
> https://www.w3.org/WAI/intro/wcag20 / (Dutch translation:
> https://www.w3.org/Translations/WCAG20-nl)
> https://www.w3.org/WAI/WCAG20/quickref
> https://www.w3.org/WAI/WCAG20/wcag2faq.html
>
> This is my first post, I did my best to be as clear as possible.
> English is not my native language so I hope this is okay translated. I
> have not added any images and code samples. I like to hear improvements for
> this post, let me know what information you extra need or miss.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla-projects/accessibility/issues/9>, or mute the
> thread
> <https://github.com/notifications/unsubscribe/ABPH8ffE2lLOT0kdO_7IonoJ3Td9cGrrks5qGzjxgaJpZM4Ip_d1>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by armenos** [comment]
Yes this is about front end template, but same issue is present in admin.
